### ISSUE TITLE
Allow tweakers to use their own class loader

### ISSUE THREAD

**Issue by SoniEx2** [Mojang/LegacyLauncher#14]
Hey! I'd like to load a tweaker in its own class loader (separate from the game code), but that doesn't seem to be possible with the current code.

I don't think there should be any changes to ITweaker. It would be nicer to be able to "inherit" the LaunchClassLoader and tweak it as needed instead.

Since game code gets loaded into the main LaunchClassLoader, I don't want to have it as the parent. I want each jar I load to have a separate LaunchClassLoader which can neither access other jars nor access the game code, except through its own IClassTransformer. (I do know how to allow APIs to work in such "sandboxed" environment, but it'd be too long for a github issue.)

Basically, each IClassTransformer would be a completely isolated unit of code, manipulating classes as needed - including other mods' classes - without being able to use those classes directly. Such coremods, as I like to call them, would be fairly lightweight compared to current Forge coremods. They would be able to access the modloader's main class, since it would get added to the main LaunchClassLoader, but that class would only be used to register the coremod-loading tweaker.

Coremods themselves would be able to register sub-coremods however they want to - in the same classloader, in a different classloader, etc. No good reason to restrict that.

You might wonder if it'd be possible to get access to the main LaunchClassLoader. And yes. But only after all coremods are registered and successfully initialized and the game is already running. And by then, it doesn't really matter that the coremod has access to the game classloader.

**Comment by SoniEx2** [comment]
TL;DR: I want the main LaunchClassLoader and all "inherited" LaunchClassLoaders to call the same tweak/transformer classes when ANY of them load a class, even tho they don't share any of those loaded classes (and the same transformer may be called multiple times for the same class, but from different class loaders).

**Comment by grum** [comment]
Please submit a pullrequest so it can be reviewed.
